Can't remove package when there is a [sources] section

I got this:
```julia
(test) pkg> rm Ipopt
ERROR: AssertionError: entry.path == path
Stacktrace:
  [1] write_env(env::Pkg.Types.EnvCache; update_undo::Bool, skip_writing_project::Bool)
    @ Pkg.Types C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\src\Types.jl:1112
  [2] write_env
    @ C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\src\Types.jl:1105 [inlined]
  [3] rm(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; mode::Pkg.Types.PackageMode)
    @ Pkg.Operations C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\src\Operations.jl:1260
  [4] rm
    @ C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\src\Operations.jl:1190 [inlined]
  [5] rm(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; mode::Pkg.Types.PackageMode, all_pkgs::Bool, kwargs::@Kwargs{io::IOContext{IO}})
    @ Pkg.API C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\src\API.jl:334
  [6] rm(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOContext{IO}, kwargs::@Kwargs{})
    @ Pkg.API C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\src\API.jl:159
  [7] rm(pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.API C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\src\API.jl:148
  [8] do_cmd(command::Pkg.REPLMode.Command, io::Base.TTY)
    @ Pkg.REPLMode C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\src\REPLMode\REPLMode.jl:407
  [9] do_cmds(commands::Vector{Pkg.REPLMode.Command}, io::Base.TTY)
    @ Pkg.REPLMode C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\src\REPLMode\REPLMode.jl:393
 [10] do_cmds(repl::REPL.LineEditREPL, commands::String)
    @ REPLExt C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\ext\REPLExt\REPLExt.jl:92
    @ REPLExt C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\ext\REPLExt\REPLExt.jl:92
 [11] on_done(s::REPL.LineEdit.MIState, buf::IOBuffer, ok::Bool, repl::REPL.LineEditREPL)
    @ REPLExt C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\ext\REPLExt\REPLExt.jl:106
 [12] #invokelatest#2
    @ .\essentials.jl:1055 [inlined]
 [13] invokelatest
    @ .\essentials.jl:1052 [inlined]
 [14] (::REPLExt.var"#47#50"{REPL.LineEditREPL})(s::REPL.LineEdit.MIState, buf::IOBuffer, ok::Bool)
    @ REPLExt C:\Julia_newest\share\julia\stdlib\v1.11\Pkg\ext\REPLExt\REPLExt.jl:128
 [15] #invokelatest#2
    @ .\essentials.jl:1055 [inlined]
 [16] invokelatest
    @ .\essentials.jl:1052 [inlined]
 [17] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit C:\Julia_newest\share\julia\stdlib\v1.11\REPL\src\LineEdit.jl:2755
 [18] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL C:\Julia_newest\share\julia\stdlib\v1.11\REPL\src\REPL.jl:1471
 [19] (::REPL.var"#75#81"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL C:\Julia_newest\share\julia\stdlib\v1.11\REPL\src\REPL.jl:480
```

Same problem here

```julia
rm Gurobi
ERROR: AssertionError: entry.repo.source == repo.source
Stacktrace:
  [1] write_env(env::Pkg.Types.EnvCache; update_undo::Bool, skip_writing_project::Bool)
    @ Pkg.Types ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Types.jl:1115
  [2] write_env
    @ ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Types.jl:1105 [inlined]
  [3] rm(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; mode::Pkg.Types.PackageMode)
    @ Pkg.Operations ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1274
  [4] rm
    @ ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1204 [inlined]
  [5] rm(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; mode::Pkg.Types.PackageMode, all_pkgs::Bool, kwargs::@Kwargs{io::IOContext{IO}})
    @ Pkg.API ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/API.jl:340
  [6] rm(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOContext{IO}, kwargs::@Kwargs{})
    @ Pkg.API ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/API.jl:159
  [7] rm(pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.API ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/API.jl:148
  [8] do_cmd(command::Pkg.REPLMode.Command, io::Base.TTY)
    @ Pkg.REPLMode ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/REPLMode/REPLMode.jl:407
  [9] do_cmds(commands::Vector{Pkg.REPLMode.Command}, io::Base.TTY)
    @ Pkg.REPLMode ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/REPLMode/REPLMode.jl:393
 [10] do_cmds(repl::REPL.LineEditREPL, commands::String)
    @ REPLExt ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/ext/REPLExt/REPLExt.jl:92
 [11] on_done(s::REPL.LineEdit.MIState, buf::IOBuffer, ok::Bool, repl::REPL.LineEditREPL)
    @ REPLExt ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/ext/REPLExt/REPLExt.jl:106
 [12] #invokelatest#2
    @ ./essentials.jl:1055 [inlined]
 [13] invokelatest
    @ ./essentials.jl:1052 [inlined]
 [14] (::REPLExt.var"#47#50"{REPL.LineEditREPL})(s::REPL.LineEdit.MIState, buf::IOBuffer, ok::Bool)
    @ REPLExt ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/ext/REPLExt/REPLExt.jl:128
 [15] #invokelatest#2
    @ ./essentials.jl:1055 [inlined]
 [16] invokelatest
    @ ./essentials.jl:1052 [inlined]
 [17] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/REPL/src/LineEdit.jl:2755
 [18] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/REPL/src/REPL.jl:1474
 [19] (::REPL.var"#75#81"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL ~/.julia/juliaup/julia-1.11.3+0.x64.linux.gnu/share/julia/stdlib/v1.11/REPL/src/REPL.jl:480
```

I have added a sources section, which links to a github repo of my own for another dependency.

If I remove the sources section manually from the `Project.toml` and call `rm Gurobi` again, everything works smoothly.

--- a/src/Types.jl
+++ b/src/Types.jl
@@ -1223,7 +1223,6 @@ function write_env(env::EnvCache; update_undo=true,
             @assert normpath(entry.path) == normpath(path)
         end
         if repo != GitRepo()
-            @assert entry.repo.source == repo.source
             if repo.rev !== nothing
                 @assert entry.repo.rev == repo.rev
             end

